qpmodel/Statis.cs

```diff
@@ -619,8 +619,12 @@ public BaseTableRef GetTargetTable()
         {
             // its child is a [gather ontop] scan
             var child = child_();
-            Debug.Assert(child is LogicGather || child is LogicScanTable);
-            if (child is LogicGather)
+
```

Comment:
do you still need to change this method since you don't invoke it anymore?

Reply:
Apparently so. The method GetTargetTable is still called from PhysicAnalyze.Open.

When the line
> var code = finalplan.Open(context);
is executed for the SQL statement
> analyze a tablesample row (15)
PhysicAnalyze.open calls
> LogicAnalyze.GetTargetTable

The type of the child node at returned by the method child_() in LogicAnalyze.GetTargetTable at this point is "logic.LogicSampleScan" so I extended the Debug.Assert.

Should the processing not follow this path for processing an Analyze statement that contains a tablesample reference?